addons/account/models/account_move.py

```diff
@@ -48,27 +48,52 @@ def _sequence_yearly_regex(self):
     def _sequence_fixed_regex(self):
         return self.journal_id.sequence_override_regex or super()._sequence_fixed_regex
 
+    @api.model
+    def _search_default_journal(self, journal_types):
+        company_id = self._context.get('force_company', self._context.get('default_company_id', self.env.company.id))
+        domain = [('company_id', '=', company_id), ('type', 'in', journal_types)]
+
+        journal = None
+        if self._context.get('default_currency_id'):
+            currency_domain = domain + [('currency_id', '=', self._context['default_currency_id'])]
+            journal = self.env['account.journal'].search(currency_domain, limit=1)
+
+        if not journal:
+            journal = self.env['account.journal'].search(domain, limit=1)
+
+        if not journal:
+            error_msg = _('Please define an accounting miscellaneous journal in your company')
+            if 'sale' in journal_types:
+                error_msg = _('Please define an accounting sale journal in your company')
+            elif 'purchase' in journal_types:
+                error_msg = _('Please define an accounting purchase journal in your company')
+            elif 'bank' in journal_types or 'cash' in journal_types:
+                error_msg = _('Please define an accounting bank or cash journal in your company')
+            raise UserError(error_msg)
+        return journal
+
     @api.model
     def _get_default_journal(self):
         ''' Get the default journal.
         It could either be passed through the context using the 'default_journal_id' key containing its id,
         either be determined by the default type.
         '''
         move_type = self._context.get('default_move_type', 'entry')
-        journal_type = 'general'
         if move_type in self.get_sale_types(include_receipts=True):
-            journal_type = 'sale'
+            journal_types = ['sale']
```

Comment:
`journal_id` is on `account.move` but I need a way to get a default "bank"/"cash" journal on my payment/statement line. So I created an helper to retrieve such default journal by types.

However, this part seems a bit messy since the last rebase..